src/docker-in-docker/install.sh

```diff
@@ -435,13 +435,14 @@ do
         eval "${dockerd_start}"
     fi
 
+    sleep 5s
+
     set +e
         docker info > /dev/null 2>&1 && docker_ok="true"
 
         if [ "${docker_ok}" != "true" ]; then
-            echo "(*) Failed to start docker, retrying in 5s..."
+            echo "(*) Failed to start docker, retrying..."
             retry_count=`expr $retry_count + 1`
-            sleep 5s
```

Comment:
reading your PR description, perhaps just lowering this delay (or making is low to start and then increase exponentially) would help?  That still provides the opportunity for this loop to exit quickly, but then also retries again in a reasonable amount of time?

Reply:
> My hypothesis is that the retry logic checks for docker status too early and triggers for a retry which brings the docker in an inconsistent state.

Mmm... The issue I am trying to fix is to ensure that we have some buffer in between https://github.com/devcontainers/features/pull/659/files#diff-a72a4e49c44e54f09feb84b7767d355d68d34fbee561d8f96a99d12d2488b9c5R431 and https://github.com/devcontainers/features/pull/659/files#diff-a72a4e49c44e54f09feb84b7767d355d68d34fbee561d8f96a99d12d2488b9c5R441.

So how would lowering this help? 🤔 
The buffer in between docker start ---> retry was 5s and it is still the same.

Reply:
Reverted!